deepgrp/__main__.py

```diff
@@ -68,127 +77,287 @@ def predict(model: tf.keras.Model, inputs: np.ndarray,
+        )
+        getattr(self, self.args.command)(self.args, options)
+
+    @staticmethod
+    def predict(args: argparse.Namespace, options: dgmodel.Options):
+        """Predict with deepgrp.
+
+        Args:
+            args (argparse.Namespace): Command line arguments.
+                Includes: 'model', 'output', 'FASTA', 'no_use_mss',
+                'step_size'
+            options (dgmodel.Options): Default and user provided options.
+        """
+
+        _LOG.debug("Loading model %s!", args.model)
+        model = tf.keras.models.load_model(
+            args.model,
+            custom_objects={
+                "ReverseComplement": dgmodel.ReverseComplement,
+            },
+        )
+        options.vecsize = model.input_shape[1]
+        _LOG.info("Model loading finished successfully!")
+        outstream = sys.stdout if args.output == "-" else open(
+            args.output, "w")
+
+        for filename in args.FASTA:
+            _LOG.info("Processing %s", filename)
+            try:
+                filestream = sys.stdin if filename == "-" else open(
+                    filename, "r")
+                for header, dnasequence in _read_multi_fasta(filestream):
+                    predictions, startpos = _predict(
+                        dnasequence,
+                        model,
+                        options,
+                        args.step_size,
+                        use_mss=not args.no_use_mss,
+                    )
+                    for segment in dgsequence.yield_segments(
+                            predictions, startpos):
+                        if segment[2] > 0:
+                            outstream.write("{}\t{}\t{}\t{}\t{}\n".format(
+                                filename, header, *segment))
+            finally:
+                if filename != "-":
+                    filestream.close()
+        if args.output != "-":
+            outstream.close()
+
+    @staticmethod
+    def train(args: argparse.Namespace, options: dgmodel.Options) -> None:
+        """Train deepgrp.
+
+        Args:
+            args (argparse.Namespace): Command line arguments.
+                Includes: 'parameter', 'trainfile', 'validfile', 'bedfile',
+                'logdir', 'modelfile'
+            options (dgmodel.Options): Default and user provided options.
+        """
+        with open(args.parameter, "r") as file:
+            parameter = dgmodel.Options.from_toml(file)
+        parameter.fromdict(options.todict())
+        logdir = args.logdir
+
+        # get which chromosome is used for training and which for validation
+        train_chr = os.path.basename(args.trainfile).split('.')[0]
+        val_chr = os.path.basename(args.validfile).split('.')[0]
+
+        # check if logdir exists?
+        if not os.path.isdir(logdir):
+            # logdir is not valid
+            logdir = os.path.join(os.getcwd(), 'logdir')
+            if not os.path.isdir(logdir):
+                os.mkdir(logdir)
```

Comment:
Maybe we misunderstood a bit here. If the logdir is not valid, we should just create this one, but not creating a different one. So maybe this one should be enough:
```python
if not os.path.isdit(logdir):
    os.mkdir(logdir)
```